tests/load/gcp/prom-query-locustfile.tpl.py

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env python3
+
+# Copyright 2021 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import json
+import random
+from datetime import datetime
+from time import time_ns
+
+from locust import HttpUser, TaskSet, constant_pacing, task
+from locust.contrib.fasthttp import FastHttpUser
+
+
+def timepad(pad: float):
+    """Decorator that guarantees a function won't return before some minimum time has elapsed.
+    Args:
+        pad: minimum time in seconds that should elapse before the wrapped function returns.
+    """
+    pad = max(0, pad)
+    def decorator(func):
+        import time
+        from functools import wraps
+        @wraps(func)
+        def inner(*args, **kwargs):
+            tic = time.time()
+            func(*args, **kwargs)
+            toc = time.time()
+            time.sleep(max(0, pad - (toc - tic)))
+        return inner
+    return decorator
+
+
+def scale_probabilities(*args):
```

Comment:
Usually such a function is called "normalise" or "normalize". Probabilities [by definition](https://en.wikipedia.org/wiki/Probability_axioms) sum to one and are never "scaled".